runtime/tests/checker/typeargument_test.go

```diff
@@ -178,6 +155,665 @@ func TestCheckTypeArguments(t *testing.T) {
 	})
 }
 
+func TestCheckParameterizedTypeIsInstantiated(t *testing.T) {
+
+	t.Parallel()
+
+	test := func(t *testing.T, code string) error {
+		baseValueActivation := sema.NewVariableActivation(sema.BaseValueActivation)
+		baseValueActivation.DeclareValue(stdlib.InclusiveRangeConstructorFunction)
+		options := ParseAndCheckOptions{
+			Config: &sema.Config{
+				BaseValueActivation: baseValueActivation,
+			},
+		}
+
+		_, err := ParseAndCheckWithOptions(t, code, options)
+		return err
+	}
+
+	t.Run("InclusiveRange<Int>", func(t *testing.T) {
+
+		t.Parallel()
+
+		err := test(t,
+			"let inclusiveRange: InclusiveRange<Int> = InclusiveRange(1, 10)",
+		)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("InclusiveRange<Int, UInt>", func(t *testing.T) {
+
+		t.Parallel()
+
+		err := test(t,
+			"let inclusiveRange: InclusiveRange<Int, UInt> = InclusiveRange(1, 10)",
+		)
+
+		errs := RequireCheckerErrors(t, err, 2)
+
+		assert.IsType(t, &sema.InvalidTypeArgumentCountError{}, errs[0])
+		assert.IsType(t, &sema.MissingTypeArgumentError{}, errs[1])
```

Comment:
I added this test case where the count of type arguments > type parameters. Ideally, we should only get one checker error i.e. the `InvalidTypeArgumentCountError`.

We get this extra error because we skip the instantiation of the parameterized type in case of invalid count which leads to the second error - https://github.com/onflow/cadence/blob/b48fd6b857c697ffdbbf16374b0fce683480b443/runtime/sema/checker.go#L2317-L2320

We report a missing type argument since we skipped instantiate call.

Reply:
Thank you for adding the test. I think this is OK. There can be situations where one semantic error could lead to multiple other errors.